src/components/AdminDashboard/AdminDashboard.js

```diff
@@ -70,117 +68,76 @@ const AdminDashboard = () => {
-  }, [pageNumber, incidentsPerPage, unapprovedIncidents]);
-
-  // incident selection (checkbox) functions
-  const selectAll = () => {
-    setAllSelected(!allSelected);
-    if (!allSelected) {
-      setSelected(currentSet.map(data => data.incident_id));
-    } else {
-      setSelected([]);
+  const approveAndRejectHandler = async (newStatus) => {
+    const currentList = getCurrentList();
+    const { selected, source } = splitIncidentsByIds(currentList, selectedIds);
+
+    // setting lat and long for approved incidents
+    if (newStatus === 'approved') {
+      for (const inc of selected) {
+        if (inc.city && inc.state) {
+          const latAndLong = await getLatAndLong(inc);
+          inc.lat = latAndLong[0];
+          inc.long = latAndLong[1];
+        }
+      }
     }
-  };
 
-  const approveAndRejectHandler = async evt => {
-    evt.preventDefault();
-    const [reviewedIncidents, unreviewedIncidents] = sortApproved(
-      currList,
-      selected
-    );
-    // MVP: One-by-one approvals. Needs work for multiple approvals simultaneously
-    const latAndLong = await getLatAndLong(reviewedIncidents[0]);
-    reviewedIncidents[0].lat = latAndLong[0];
-    reviewedIncidents[0].long = latAndLong[1];
-
-    putIncidents(oktaAxios, reviewedIncidents, confirmStatus);
-
-    if (listType === 'unapproved') {
-      setUnapprovedIncidents(unreviewedIncidents);
-    } else if (listType === 'form-responses') {
-      //did 'else if' incase we wanted to add Reject button to Approved list
-      setFormResponses(unreviewedIncidents);
-    }
-    setAllSelected(false);
-    setSelected([]);
-    setConfirmApprove(false);
-    setConfirmReject(false);
-    if (pageNumber > lastPage) {
-      setPageNumber(pageNumber - 1);
-    }
-  };
+    putIncidents(oktaAxios, selected, newStatus)
+      .then(res => {
+        console.log(res);
 
-  // toggling confirmation of approve/reject buttons
-  const confirmApproveHandler = evt => {
-    evt.preventDefault();
-    setConfirmApprove(true);
-    setConfirmStatus('approved');
-  };
+        if (listType === 'pending') {
+          setPendingIncidents(source);
+        }
 
-  const confirmRejectHandler = evt => {
-    evt.preventDefault();
-    setConfirmReject(true);
-    setConfirmStatus('rejected');
-  };
+        else if (listType === 'form-responses') {
+          setFormResponses(source);
+        }
 
-  const confirmCancel = evt => {
-    evt.preventDefault();
-    setConfirmApprove(false);
-    setConfirmReject(false);
-    setConfirmStatus('pending');
-  };
+        else if (listType === 'approved') {
+          setApprovedIncidents(source);
+        }
 
-  const handlePerPageChange = evt => {
-    evt.preventDefault();
-    setIncidentsPerPage(Number(evt.target.value));
-    setPageNumber(1);
+        setAllSelected(false);
+        setSelectedIds([]);
+      })
+      .catch(err => {
+        console.log(err);
+
+        // TODO: Better error handling!
+      });
   };
 
```

Comment:
Same concern as above with last 4 functions removed, just unaware of why they were removed / how that functionality is reproduced, but awaiting loom walkthrough to see

Reply:
These are the event handlers for the approve/reject (and then the following confirm/cancel), and are now in the IncidentStatus component.